perfkitbenchmarker/scripts/object_storage_api_tests.py

```diff
@@ -266,57 +346,74 @@ def WriteObjects(storage_schema, bucket, object_prefix, count,
         successfully written.
     host_to_connect: An optional endpoint string to connect to.
   """
-  payload_bytes = bytearray(size)
-  # Use a while loop to fill up the byte array which is by default 100MB in size
-  # don't use a for...range(100M), range(100M) will create a list of 100mm items
-  # with each item being the default object size of 30 bytes or so, it will lead
-  # to out of memory error.
-  i = 0
-  while i < size:
-    payload_bytes[i] = ord(random.choice(string.letters))
-    i += 1
 
-  payload_string = payload_bytes.decode('ascii')
+  payload_bytes, payload_string = GenerateWritePayload(size)
 
-  for i in range(count):
+  for i in xrange(count):
     object_name = '%s_%d' % (object_prefix, i)
 
-    # Ready to go!
-    start_time = time.time()
-
-    # We write the object to storage provider and if successful we place the
-    # name of the object to the objects_written list. When a write fails, the
-    # object won't be placed into the objects_written list, and then it's up to
-    # the caller to decide if they want to tolerate the failure and accept
-    # the results, depending on test scenarios.
     try:
-      if _useBotoApi(storage_schema):
-        object_path = '%s/%s' % (bucket, object_name)
-        object_uri = boto.storage_uri(object_path, storage_schema)
-        if host_to_connect is not None:
-          object_uri.connect(host=host_to_connect)
-
-        object_uri.set_contents_from_string(payload_string)
-      else:
-        _AZURE_BLOB_SERVICE.put_block_blob_from_bytes(bucket, object_name,
-                                                      bytes(payload_bytes))
-
-      latency = time.time() - start_time
+      _, latency = WriteObjectFromBuffer(
+          storage_schema, bucket, object_prefix,
+          payload_bytes, payload_string, size,
+          host_to_connect=host_to_connect)
 
+      objects_written.append(object_name)
       if latency_results is not None:
         latency_results.append(latency)
-
       if bandwidth_results is not None and latency > 0.0:
         bandwidth_results.append(size / latency)
+    except Exception as e:
+      logging.info('Caught exception %s while writing object %s' %
+                   (e, object_name))
 
-      objects_written.append(object_name)
-    except:
-      logging.exception('Caught exception while writing object %s.',
-                        object_name)
+
+def WriteObjectFromBuffer(storage_schema, bucket, object,
+                          payload_bytes, payload_string, size,
```

Comment:
Renamed "object" -> "object_name", "bucket" -> "bucket_name" and "size" -> "size_to_write". I think those were the ambiguous ones.